air/src/stack/system_ops/mod.rs

```diff
@@ -0,0 +1,134 @@
+use crate::utils::are_equal;
+use ::vm_core::{FieldElement, FMP_COL_IDX, STACK_TRACE_OFFSET};
+use winter_air::TransitionConstraintDegree;
+
+#[cfg(test)]
+pub mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+/// The number of unique transition constraints in system operations.
+pub const NUM_CONSTRAINTS: usize = 2;
+
+/// The degrees of constraints in individual system operations of the system ops.
+pub const CONSTRAINT_DEGREES: [usize; NUM_CONSTRAINTS] = [
+    // Given it is a degree 7 operation, 7 is added to all the individual constraints
+    // degree.
+
+    // There's one unique constraint in NOOP system operation in which the first element of the
+    // stack is copied over onto the next frame.
+    8,
+    // There's one unique constraint in FMPADD system operation in which the top element in the
+    // stack is is incremented by fmp register value.
+    8,
+];
+
+// SYSTEM OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the system operations.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    CONSTRAINT_DEGREES
+        .iter()
+        .map(|&degree| TransitionConstraintDegree::new(degree))
+        .collect()
+}
+
+/// Returns the number of transition constraints for the system operations.
+pub fn get_transition_constraint_count() -> usize {
+    NUM_CONSTRAINTS
+}
+
+/// Enforces constraints for the system operations.
+pub fn enforce_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &OpFlags<E>,
+) -> usize {
+    let mut index = 0;
+
+    // fetching the respective flags of noop and fmpadd from op_flag struct.
+    let noop_op_flag = op_flag.noop();
+    let fmpadd_op_flag = op_flag.fmpadd();
+
+    // Enforce unique constaints of the SWAP operation.
+    index += enforce_noop_unique_constraints(frame, result, noop_op_flag);
+
+    index += enforce_fmpadd_unique_constraints(frame, &mut result[index..], fmpadd_op_flag);
+
+    index
+}
+
+// TRANSITION CONSTRAINT HELPERS
+// ================================================================================================
+
+/// Enforces unique constraints of the NOOP operation. The NOOP operation does nothing to the stack
+/// stack and the entire stack state is copied over in the next cycle.
+/// - Enforces that all the elements in the next cycles are the same as in current cycle.
+pub fn enforce_noop_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: E,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // Enforces the first element in the next frame is same as first element in the current frame.
+    result[constraint_offset] = op_flag * are_equal(frame.selector(0), frame.selector_next(0));
```

Comment:
Do we need this constraint at all? I think this will be enforced as a part of general stack constraints which we'll tackle in the next PR.

Reply:
Removed it in the latest PR